docs/how_pool_works.md

```diff
@@ -0,0 +1,198 @@
+This document is an introduction to understanding how the funding pool contract works.
+It doesn't describe all the implementation details or the API,
+it focuses on the basic principles of the core mechanics.
+
+# Overview
+
+The funding pool is a smart contract which creates real-time streams of donations.
+One can start, alter or end the process of sending their funds at any time with immediate effect.
+The flow of funds is automatically maintained and steady over time.
+
+There are 3 roles present in the contract.
+Any Ethereum address can simultaneously take on any of these roles.
+
+- **The sender**: has assets and chooses who do they want to send them to,
+how much, and at what rate
+- **The receiver:** receives funds from senders
+- **The proxy**: receives funds from senders,
+but immediately passes them to receivers of their choice
+
+## The cycles
+
+The whole blockchain history is divided into cycles of equal length.
+Every cycle contains the same, constant number of blocks.
+In the examples below, we assume that the cycle length is 5.
+
+![](how_pool_works_1.png)
+
+# The receiver
+
+There are no setup steps for one to become a receiver.
+Any address can receive donations at any time, from any sender.
+The only function of this role is collection of funds sent by others.
+
+## The deltas
+
+On every block, a sender can send funds to a receiver.
+The receiver can collect funds sent on a given block only when the cycle containing it is over.
```

Comment:
We should elaborate on this
* The “collect” transaction triggered by a receiver moves all collectable funds to the recipients account
* We need to explain how the collectable funds are calculated
* We need to explain that collecting funds reduces the collectable funds.

Reply:
Fixed